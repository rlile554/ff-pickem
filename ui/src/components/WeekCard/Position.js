import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { FiLock } from 'react-icons/fi';

const Container = styled.div`
    width: 100%;
    height: 50px;
    border-top: 1px solid #979797;
    display: flex;
    align-items: center;
    padding-left: 2px;
`;

const PosDiv = styled.div`
    height: 40px;
    width: 40px;
    border-radius: 50%;
    background-color: #5b5b5b;
    display: flex;
    align-items: center;
    justify-content: center;
    color: white;
    font-weight: bold;
`;

const IconDiv = styled.div`
    width: 25px;
    height: 25px;
    font-size: 20px;
    padding-top: 5px;
`;

const Name = styled.div`
    font-weight: bold;
    width: 180px;
    text-align: center;
`;

const Points = styled.div`
    font-weight: bold;
    text-align: right;
    width: 60px;
`;

const Position = props => {
    const { position, locked, name, points } = props;
    return (
        <Container>
            <PosDiv id="position">{position}</PosDiv>
            <IconDiv id="iconDiv">{locked && <FiLock />}</IconDiv>
            <Name id="name">{name}</Name>
            <Points id="points">{points}</Points>
        </Container>
    );
};

Position.propTypes = {
    //The position of the player
    position: PropTypes.string,
    //wether or not the pick is locked in
    locked: PropTypes.bool,
    //the name of the player
    name: PropTypes.string,
    //the numer of ponts the player currently has
    points: PropTypes.number
};

export default Position;
