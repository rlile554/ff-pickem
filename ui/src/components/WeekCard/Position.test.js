import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Position from './Position';

describe('Position', () => {
    const props  = {
        name: "Bob Guy",
        position: "QB",
        locked: true,
        points: 123
    };

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Position {...props} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('to match snapshot', () => {
        const component = renderer.create(<Position {...props} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders name prop', () => {
        const component = renderer.create(<Position {...props} />);
        let tree = component.toJSON();
        const nameDiv = tree.children.find(item => item.props.id === 'name');
        expect(nameDiv.children[0]).toEqual(props.name);
    });

    it('renders position prop', () => {
        const component = renderer.create(<Position {...props} />);
        let tree = component.toJSON();
        const nameDiv = tree.children.find(item => item.props.id === 'position');
        expect(nameDiv.children[0]).toEqual(props.position);
    });

    it('renders points prop', () => {
        const component = renderer.create(<Position {...props} />);
        let tree = component.toJSON();
        const nameDiv = tree.children.find(item => item.props.id === 'points');
        expect(nameDiv.children[0]).toEqual(props.points.toString());
    });

    it('shows the lock icon when the locked prop is true', () => {
        const component = renderer.create(<Position {...props} />);
        let tree = component.toJSON();
        const nameDiv = tree.children.find(item => item.props.id === 'iconDiv');
        expect(nameDiv.children.length).toEqual(1); 
    });

    it('does not shows the lock icon when the locked prop is false', () => {
        const component = renderer.create(<Position />);
        let tree = component.toJSON();
        const nameDiv = tree.children.find(item => item.props.id === 'iconDiv');
        expect(nameDiv.children).toBeNull();
    })
});