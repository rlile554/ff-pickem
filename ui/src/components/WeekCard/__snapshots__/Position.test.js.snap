// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Position to match snapshot 1`] = `
<div
  className="sc-bdVaJa hvvqAj"
>
  <div
    className="sc-bwzfXH fzysLj"
    id="position"
  >
    QB
  </div>
  <div
    className="sc-htpNat klVHJR"
    id="iconDiv"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="11"
        rx="2"
        ry="2"
        width="18"
        x="3"
        y="11"
      />
      <path
        d="M7 11V7a5 5 0 0 1 10 0v4"
      />
    </svg>
  </div>
  <div
    className="sc-bxivhb bfViTE"
    id="name"
  >
    Bob Guy
  </div>
  <div
    className="sc-ifAKCX jlmnUT"
    id="points"
  >
    123
  </div>
</div>
`;
